### PR TITLE
Added test cases for create api

### PR DESCRIPTION
**Pull request by hrithik2200**

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -87,27 +87,27 @@ def index():
...
         promotion.create()
         message = promotion.serialize()
-        location_url = url_for("get_promotion", promotion_id=promotion.promotion_id, _external=True)
-        return jsonify(message), status.HTTP_201_CREATED, {"Location": location_url}
+        # location_url = url_for("get_promotion", promotion_id=promotion.promotion_id, _external=True)
+        return jsonify(message), status.HTTP_201_CREATED
     except DataValidationError as error:
         abort(status.HTTP_400_BAD_REQUEST, str(error))
-    except Exception as error:  # pylint: disable=broad-except
+    except Exception as error:
```
Comment (ChungLinWu0829) [inline service/routes.py:108]: I think we should include this comment to disable the specific lint error regarding broad exceptions.

File: service/routes.py [hunk 2]
```diff
@@ -160,3 +160,9 @@ def abort_with_error(error_code, error_msg):
     """
     app.logger.error(error_msg)
     abort(error_code, error_msg)
+
```
Comment (ChungLinWu0829) [inline service/routes.py:163]: @mingyi850 has different implementations for handling the internal server error in this PR(https://github.com/CSCI-GA-2820-SU24-001/promotions/pull/19/files), I think we need to agree on one consistent approach for our error handling. Could you two align on this?

**Comment by ChungLinWu0829** [comment]
Might be an issue with my setup. These two test cases failed in my environment. Are they working correctly on your end?
image

File: service/routes.py (hunk 1 above)
Comment (hrithik2200) [reply service/routes.py:108]: Okay, sure

**Review by anirudh28 — COMMENTED** [review]
Looks good.

File: service/models.py [hunk 3]
```diff
@@ -61,19 +61,32 @@ class Promotion(db.Model):  # pylint: disable=too-many-instance-attributes
...
+    # promotion_type = db.Column(db.Enum(PromotionType), nullable=False)
+    # promotion_scope = db.Column(db.Enum(PromotionScope))
+    # start_date = db.Column(db.DateTime, nullable=False)
+    # end_date = db.Column(db.DateTime)
+    # promotion_value = db.Column(db.Double, nullable=False)
+    # promotion_code = db.Column(db.String(63), nullable=True)
+    # created_by = db.Column(db.Uuid)
+    # modified_by = db.Column(db.Uuid, nullable=True)
+    # created_when = db.Column(db.DateTime)
+    # modified_when = db.Column(db.DateTime)
```
Comment (mingyi850) [inline service/models.py:76]: Remove comments

File: service/routes.py [hunk 4]
```diff
@@ -87,27 +87,27 @@ def index():
...
+    check_content_type("application/json")
     data = request.get_json()
     promotion = Promotion()
     try:
         promotion.deserialize(data)
         promotion.create()
         message = promotion.serialize()
-        location_url = url_for("get_promotion", promotion_id=promotion.promotion_id, _external=True)
-        return jsonify(message), status.HTTP_201_CREATED, {"Location": location_url}
+        # location_url = url_for("get_promotion", promotion_id=promotion.promotion_id, _external=True)
```
Comment (mingyi850) [inline service/routes.py:104]: remove comments

File: service/routes.py (hunk 1 above)
Comment (mingyi850) [inline service/routes.py:108]: Use error handlers in error_handlers.py

File: service/routes.py [hunk 5]
```diff
@@ -160,3 +160,9 @@ def abort_with_error(error_code, error_msg):
     """
     app.logger.error(error_msg)
     abort(error_code, error_msg)
+    
+def check_content_type(content_type):
+    """Checks that the media type is correct"""
+    if request.headers["Content-Type"] != content_type:
+        app.logger.error("Invalid Content-Type: %s", request.headers["Content-Type"])
+        abort(status.HTTP_415_UNSUPPORTED_MEDIA_TYPE, f"Content-Type must be {content_type}")
```
Comment (mingyi850) [inline service/routes.py:168]: Not needed as we discussed use error handler in error_handler.py

**Review by mingyi850 — CHANGES_REQUESTED** [review]
Some changes needed

**Comment by mingyi850** [comment]
service/common/datetime_utils.py:34:1: W293 blank line contains whitespace
service/routes.py:112:1: E302 expected 2 blank lines, found 1
service/routes.py:163:1: W293 blank line contains whitespace
service/routes.py:164:1: E302 expected 2 blank lines, found 1
tests/test_models.py:192:1: W293 blank line contains whitespace
tests/test_models.py:196:1: W293 blank line contains whitespace
tests/test_models.py:197:46: E231 missing whitespace after ','
tests/test_models.py:197:69: W291 trailing whitespace
1     E231 missing whitespace after ','
2     E302 expected 2 blank lines, found 1
1     W291 trailing whitespace
4     W293 blank line contains whitespace
8
make: *** [Makefile:36: lint] Error 1

Please fix lint issues too

**Review by mingyi850 — APPROVED** [review]
LGTM

**Pull request merged**